cluster-operator/src/main/java/io/strimzi/operator/cluster/PlatformFeaturesAvailability.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster;
+
+import io.vertx.core.json.Json;
+
+import java.util.Properties;
+
+public class PlatformFeaturesAvailability {
+    private Properties properties;
+
+    static final String MAJORVERSION = "major";
+    static final String MINORVERSION = "minor";
+    static final String ISOPENSHIFT = "isOpenshift";
+    static final String NETWORKPOLICIES = "networkPolicies";
+
+    static final String KUBERNETES1_8 = "1.8";
+    static final String KUBERNETES1_9 = "1.9";
+    static final String KUBERNETES1_10 = "1.10";
+    static final String KUBERNETES1_11 = "1.11";
+
+    public PlatformFeaturesAvailability(boolean isOpenshift, String versionResponse) {
```

Comment:
`versionResponse` has to be a certain JSON string for this to work. Ask yourself why you put the JSON parsing and unpicking logic here, in the constructor, rather than in `Main.isOnOpenShift()`. That's the place which knows it's talking to an API server and that the API server will give a particular JSON response. So that's the place which should deal with the JSON. It should call a `PlatformFeaturesAvailability` constructor which just takes the `boolean`, `int` etc fields which we're actually interested in.

This also simplifies tests too: The tests should not need to know the particular kind of JSON which is needed (only tests for `Main.isOnOpenShift()` should know that). But the constructor doesn't take the JSON string as a parameter then tests don't need to know, they can just pass `true, 1, 11` etc as arguments.